### PR TITLE
#7995 Micro-ROS configuration NuttX tutorial

### PR DESCRIPTION
**Pull request by jfm92**
This tutorial shows a little guide of how to configure NuttX to include Micro-ROS on a new board.

### CODE REVIEW

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ros2 run micro_ros_setup configure_firmware.sh nsh
+```
+
+If everything goes fine, it should output the next message:
+```bash
+Copy files
+Refreshing...
+```
+
+Now is set the basic NSH configuration. On the forward steps, we will set the required configuration to run Micro-ROS on this board.
```
Comment (FranFin) [inline _docs/tutorials/advanced/nuttx/add_microros_config/index.md:64]:
```suggestion
Now the basic NSH configuration is set. With the steps that follow, we will set the required configuration to run Micro-ROS on this board.
```

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+## Micro-ROS configuration.
+
+On the previous console execute the next commands:
```
Comment (FranFin) [inline _docs/tutorials/advanced/nuttx/add_microros_config/index.md:68]:
```suggestion
In the same console as before, run the commands:
```

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+```bash
+cd firmware/NuttX
+make menuconfig
+```
+
+This should show this menu:
```
Comment (FranFin) [inline _docs/tutorials/advanced/nuttx/add_microros_config/index.md:74]:
```suggestion
The menu below should appear:
```

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+![](images/nuttx_menuconfig.png)
+
+Note: Remember that you should follow this previous tutorial before starting it. [Adding Micro-ROS to a NuttX board configuration](https://micro-ros.github.io/docs/tutorials/advanced/nuttx/microros_nuttx_bsp/)
```
Comment (FranFin) [inline _docs/tutorials/advanced/nuttx/add_microros_config/index.md:77]: You already said it before. Are you sure it makes sense to repeat it here? Maybe the best thin to do is to remove it from here, and also from above, and create instead a 'Disclaimer' in bold above in a new line, close to where the other Disclaimer is. (I prefer the format where the disclaimers are not section but something inline like "Disclaimer!")

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+On the menu, you need to set the next configuration.
```
Comment (FranFin) [inline _docs/tutorials/advanced/nuttx/add_microros_config/index.md:79]:
```suggestion
In the menu, you need to set the following configuration.
```

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+- Library Routines > Standard Math Library
+- Library Routines > sizeof(_Bool) is a 8-bits
+- Library Routines > Build uClibc++ (must be installed) 
+- Application Configuration > micro-ROS
+- Application Configuration > micro-ROS > Transport > Serial Transport
+- Application Configuration > Examples > microROS Publisher
+
+Push two times the key "ESC" and set yes, when it asks you if you want to save it.
+
+Now the configuration is properly set, is just left to compile it. To do so, continue on the same console and execute the commands:
```
Comment (FranFin) [inline _docs/tutorials/advanced/nuttx/add_microros_config/index.md:93]:
```suggestion
Now the configuration is properly set, the only thing left is to compile it. To do so, continue in the same console and execute the commands:
```

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+```
+cd ../..
+ros2 run micro_ros_setup build_firmware.sh
+```
+
+If everything goes fine, it should return the next output:
```
Comment (FranFin) [inline _docs/tutorials/advanced/nuttx/add_microros_config/index.md:99]:
```suggestion
If everything goes fine, it should return the following output:
```

File: _docs/tutorials/advanced/nuttx/add_microros_config/index.md (hunk 4 above)
Comment (jfm92) [reply _docs/tutorials/advanced/nuttx/add_microros_config/index.md:77]: Done!

**Pull request merged**